### DISCUSSION TITLE
Best way to access Rust struct in Slint?

### DISCUSSION THREAD

**Discussion by richardmcatlin** [slint-ui/slint#4252] [Q&A]
I have rust structs coming from the backend (aka Clean Architecture Entities).  What's the best way to access these structs from slint?  In the examples, I see slint defining the struct and rust using them, but not the other way around.  Thanks.

**Comment by Vadoola** [comment]
I'll have to dig around to find it but there should be some issues and / or discussions about this. Unless it changed recently (and I missed it) this currently isn't possible,  but they are working on it from what I recall.

Edit: here is an issue for the same basic request https://github.com/slint-ui/slint/issues/1726

**Comment by ogoffart** [comment]
While waiting that #1726 is supported, i recommend you just implement manually the `From` rust trait to convert between the Rust and an equivalent Slint struct, and convert between these.

**Reply by Vadoola** [reply]
Good point, I forgot to mention this, but it's exactly what I'm doing in Tomotroid for the settings and theme loading.
